### ISSUE TITLE
Use of undeclared type or module `ErrorKind` from nom-4 macros

### ISSUE THREAD

**Issue by djc** [Geal/nom#759]
- Rust version: 1.26.0
- nom version: 4.0
- nom compilation features used: default

This test case:

```
#[macro_use]
extern crate nom;

use std::str;

fn atom_specials(c: u8) -> bool {
    true
}

named!(capability<&str>, do_parse!(
    tag_s!(" ") >>
    atom: map_res!(take_till1_s!(atom_specials), str::from_utf8) >>
    ("a")
));
```

results in:

```
error[E0433]: failed to resolve. Use of undeclared type or module `ErrorKind`
  --> src/main.rs:9:1
   |
9  | / named!(capability<&str>, do_parse!(
10 | |     tag_s!(" ") >>
11 | |     atom: map_res!(take_till1_s!(atom_specials), str::from_utf8) >>
12 | |     ("a")
13 | | ));
   | |___^ Use of undeclared type or module `ErrorKind`
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

**Comment by lukaslueg** [comment]
There is a similar issue with `nom::be_u16`  and it's variants: If `Endianess` is not brought in scope manually, they fail to compile due to `undeclared type`. This is particularly annoying because I like my namespaces to be clean and only have one central `use nom` at the top and fully qualified paths everywhere else.

**Comment by Geal** [comment]
@lukaslueg with 2769946 the issue with `Endianness` should be fixed too

**Comment by Geal** [comment]
merged #760
